lib/pluggable.js

```diff
@@ -0,0 +1,115 @@
+// TODO: make DRY
+
+import * as most from "most";
+
+
+export const CONTINUE = Symbol.for("interlock.continue");
+
+
+function getContext (baseCxt, dependencies) {
+  const context = Object.create(null);
+  Object.assign(context, baseCxt);
+  for (let depKey of Object.keys(dependencies)) {
+    let dep = dependencies[depKey];
+    context[depKey] = dep.__isPluggable__ ? dep.bind(baseCxt) : dep;
```

Comment:
The `baseCxt` is passed down from parent to child, all the way down the call stack.  The actual context that is applied to the function itself (the return value here), incorporates both the `baseCxt` and a single property for each dependency that is defined (also bound to the `baseCxt`).  This allows data to be shared throughout the application without concerns that a mutation in one function's context will effect another, or that the dependencies of some parent function will pollute the context of its siblings, parents, or children.

The abstraction itself is a bit opaque, but the consuming experience is very good.  Everything looks and behaves (and can be tested) as pure functions.  If the `Pluggable` function is invoked without a context, it behaves just like a normal function (not `Pluggable`).  This way it can be composed by external tools or build scripts, and allows configuration to be shared without pain.

It also makes it trivial to extend the `Pluggable` functions through extensions.  I'm not done with user-facing API for plugins yet, but the tests show how this will work internally.  It is possible to override the output of a `Pluggable` function directly, and it is also possible to transform the output after the fact.